src/test/java/org/jabref/logic/util/BracketedExpressionExpanderTest.java

```diff
@@ -0,0 +1,153 @@
+package org.jabref.logic.util;
+
+import javafx.embed.swing.JFXPanel;
+
+import org.jabref.model.database.BibDatabase;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.BibtexEntryTypes;
+import org.jabref.model.entry.BibtexString;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class BracketedExpressionExpanderTest {
+    // The javafxPanel instace is only created to suppress the ugly
```

Comment:
That is indeed quite ugly. As far as I can see this comes from the using the logger. It's a little weird, because I am almost sure that the logger didn't trigger these exceptions earlier, but this is probably not caused by this PR.

I think I would prefer it if you removed the JFXPanel here. It is better to have exceptions written into the console (the tests work anyway) than to have dependencies such as this in the code. But maybe another developer might disagree with me here.

Reply:
Removed JFXPanel, but added FIXMEs to the tests that trigger the exception logging -- we should find out what goes on there. The exceptions can not be caught in the test case code (!), apparently meaning that they are thrown and logged in a separate thread (JUnut?).

Only two test cases, unbalancedBracketExpansionTest and unbalancedLastBracketExpansionTest, cause this behaviour, since only they use the logger to report syntax errors. Another solution would be to drop logger dependency from BracketedExpressionExpander and throw exceptions on sytax errors, expecting that the caller catches and logs them appropriately. I would like this approach more, but currently there is a risk that uncaught exceptions will crash the application or produce obscure notifications.